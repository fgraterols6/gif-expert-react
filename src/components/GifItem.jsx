import PropTypes  from 'prop-types';

export const GifItem = ({ title, url, id}) => {

    return (
        <img src={url} alt={title} width="100" height="100" />     
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}