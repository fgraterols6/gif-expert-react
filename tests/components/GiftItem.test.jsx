import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GiftItem />', () =>Â {
    const title = 'Taylor Swift';
    const url = 'https://taylor-swift-1.com/img-1.png'


    test('should be match with the snapshot', () => {

        const { container } = render(<GifItem title={ title } url={ url } />)
        expect( container ).toMatchSnapshot();
    });

    test('should show the image with the URL and ALT', () => {


        render(<GifItem title={ title } url={ url } />)
        // screen.debug()
        // expect(screen.getByRole('img').src).toBe( url );

        const {src, alt} = screen.getByRole('img')

        expect(src).toBe(url)
        expect(alt).toBe(title)
    });

});