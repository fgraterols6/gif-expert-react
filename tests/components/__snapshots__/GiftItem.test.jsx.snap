// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem /> should be match with the snapshot 1`] = `
<div>
  <img
    alt="Taylor Swift"
    height="100"
    src="https://taylor-swift-1.com/img-1.png"
    width="100"
  />
</div>
`;
